
From: Mike Turquette <mturquette@ti.com>
Date: Mon, 23 May 2011 18:24:15 +0000 (-0500)
Subject: OMAP4: CPUfreq: manage DPLL cascading constraint
X-Git-Url: http://git.omapzoom.org/?p=kernel%2Fomap.git;a=commitdiff_plain;h=b81c60592ea9498400d6de87d9d840d505e78fec

OMAP4: CPUfreq: manage DPLL cascading constraint

Only release a DPLL cascading constraint when at the lowest OPP, otherwise
hold a constraint blocking DPLL cascading entry.

Change-Id: Ie757115df332a76f5f0821977ee2d95d1fc64e36
Signed-off-by: Mike Turquette <mturquette@ti.com>
---

Index: kernel/arch/arm/plat-omap/cpu-omap.c
===================================================================
--- kernel.orig/arch/arm/plat-omap/cpu-omap.c	2011-05-28 00:29:31.079637585 +0900
+++ kernel/arch/arm/plat-omap/cpu-omap.c	2011-05-28 00:30:01.409598345 +0900
@@ -36,6 +36,7 @@
 #if defined(CONFIG_ARCH_OMAP3) || defined(CONFIG_ARCH_OMAP4)
 #include <plat/omap-pm.h>
 #include <plat/opp.h>
+#include <mach/omap4-common.h>
 #endif
 
 #define VERY_HI_RATE	900000000
@@ -157,9 +158,16 @@
 	}
 #endif
 
+	if (cpu_is_omap44xx() && target_freq > policy->min)
+		dpll_cascading_blocker_hold(mpu_dev);
+
 	freq = target_freq * 1000;
 	if (opp_find_freq_ceil(mpu_dev, &freq))
 		omap_device_set_rate(mpu_dev, mpu_dev, freq);
+
+	if (cpu_is_omap44xx() && target_freq == policy->min)
+		dpll_cascading_blocker_release(mpu_dev);
+
 #ifdef CONFIG_SMP
 	/*
 	 * Note that loops_per_jiffy is not updated on SMP systems in
